destination.go

```diff
@@ -81,7 +81,11 @@ func (d *Destination) Flush(context.Context) error {
 func (d *Destination) Teardown(ctx context.Context) error {
 	sdk.Logger(ctx).Info().Msg("Tearing down a Kafka Destination...")
 	if d.Producer != nil {
-		_ = d.Producer.Close()
+		err := d.Producer.Close()
+		if err != nil {
+			return fmt.Errorf("failed closing Kafka producer: %w", err)
+		}
+		return nil
```

Comment:
I'd remove this line, in case we add more teardown logic.

Reply:
Makes sense. :+1: